sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -2502,6 +2502,26 @@ class AdaptiveQueryExecSuite
     }
   }
 
+  test("SPARK-48037: Fix SortShuffleWriter lacks shuffle write related metrics " +
+    "resulting in potentially inaccurate data") {
+    withTable("t3") {
+      withSQLConf(
+        SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
+        SQLConf.SHUFFLE_PARTITIONS.key -> "16777217") {
+        sql("CREATE TABLE t3 USING PARQUET AS SELECT id FROM range(2)")
+        val (plan, adaptivePlan) = runAdaptiveAndVerifyResult(
+          """
+            |SELECT id, count(*)
+            |FROM t3
+            |GROUP BY id
+            |LIMIT 1
+            |""".stripMargin, skipCheckAnswer = true)
```

Comment:
do we have to skip checking the answer? I think the query can be `LIMIT 10` so that the result is deterministic.

Reply:
AFAIK the `checkAnswer` util will sort the data before comparison.

Reply:
It does sort, but the final result is sorted locally. Because the result of the limit is uncertain, the local sort has no effect.

org.apache.spark.sql.QueryTest#getErrorMessageInCheckAnswer
```
val sparkAnswer = try df.collect().toSeq
```

Reply:
we can make it certain if the limit is larger than the number of result rows?

Reply:
In this test case, because `spark.sql.adaptive.enabled=false`, partitions will not be merged. 
It has a large number of partitions and tasks, so it requires a large amount of driver memory to execute successfully.